flax/core/lift.py

```diff
@@ -447,14 +446,14 @@ def jitted(variable_groups_xs, rng_groups_xs, *args):
 
     return jitted(variable_groups_xs, rng_groups_xs, *args)
 
-  return pack(inner, (in_variables,), (out_variables,), (rngs,))
+  return pack(inner, (variables,), (variables,), (rngs,))
 
 
 def remat_scan(body_fn: Callable[..., Any], scope: Scope, carry: Any,
                lengths: Sequence[int],
-               variable_carry: KindFilter = False,
-               variable_axes: Mapping[KindFilter, InOutScanAxis] = {},
-               split_rngs: Mapping[KindFilter, bool] = {}):
+               variable_carry: CollectionFilter = False,
+               variable_axes: Mapping[CollectionFilter, InOutScanAxis] = {},
+               split_rngs: Mapping[RNGFilter, bool] = {}):
```

Comment:
It's cool to have these indeed be different types (even if they're the same under the hood) for sake of self-documenting these functions.

Reply:
I agree, there's a weird balancing act between keeping things consistent and abstract in the implementation and separated in the public api for clarity.